hdwallet/src/main/kotlin/io/provenance/hdwallet/wallet/Wallet.kt

```diff
@@ -79,31 +92,72 @@ interface Wallet {
          * @param hrp The human-readable prefix to use when generating bech32 addresses from this wallet.
          * @param passphrase The passphrase to use when generating the seed that will be used to create this wallet.
          * @param mnemonicWords The BIP39 mnemonic phrase to use to generate the seed for this wallet.
-         * @param testnet A flag that specifies if this wallet corresponds to a test net. If omitted, the default
-         * `false` will be used.
+         * @param publicKeyOnly Flag specifying to generate only public keys [default: false]
+         * @param testnet Flag specifying if this wallet corresponds to a test net. [default: false]
          * @return [Wallet]
          */
         fun fromMnemonic(
             hrp: String,
             passphrase: String,
             mnemonicWords: MnemonicWords,
-            testnet: Boolean = false
+            publicKeyOnly: Boolean = false,
+            testnet: Boolean = false,
+            curve: Curve = CURVE,
         ): Wallet = fromMnemonic(
             hrp = hrp,
             passphrase = passphrase.toCharArray(),
             mnemonicWords = mnemonicWords,
-            testnet = testnet
+            publicKeyOnly = publicKeyOnly,
+            testnet = testnet,
+            curve = curve,
         )
     }
 }
 
+/**
```

Comment:
Thanks for this documentation. Super super helpful

Reply:
👍